src/nvim/drawline.c

```diff
@@ -2546,7 +2470,6 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool number_onl
         && wp->w_p_list
         && (wp->w_p_wrap ? (wp->w_skipcol > 0 && wlv.row == 0) : wp->w_leftcol > 0)
         && wlv.filler_todo <= 0
-        && wlv.draw_state > WL_STC
```

Comment:
This makes 'listchars' "precedes" draw after 'breakindent' and 'showbreak' (overlapping the buffer text) when 'smoothscroll' is set. Previously it draws over 'breakindent' or 'showbreak'. Not sure if this is really a problem.

Reply:
But maybe 'listchars' "precedes" can be changed to draw over text already drawn, like what is currently done for 'smoothscroll' marker.